fw/src/mppt/global/particleswarm.hpp

```diff
@@ -0,0 +1,395 @@
+/**
+ * @file particleswarm.hpp
+ * @author Afnan Mir (afnanmir@utexas.edu) & Nash Wu
+ * @brief Implementation of Particle Swarm Optimization GlobalMPPTAlgortihm.
+ * @version 0.2.0
+ * @date 2024-01-27
+ * 
+ * @copyright Copyright (c) 2024
+ * 
+ */
+
+#include <random>
+
+#pragma once
+#include "../mppt.hpp"
+#include "../local/pando.hpp"
+
+#define    NUM_AGENTS 4
+#define    C1 0.02
+#define    C2 0.5
+#define    W 0.35
+
+class Particle final {
+
+    public:
+
+        Particle();
+
+        Particle(float xPos, float vel) {
```

Comment:
highly recommend documentation on any new functions introduced in the hpp or cpp files
see mppt.hpp as an example

also the hpp file really only has implementation code if it is (1) implementing a small and concise set of virtual or concrete methods from another class (2) the implementation is very short and encapsulated.

I think at this point you will want to integrate this into a dedicated .cpp file.